pandas/core/frame.py

```diff
@@ -2576,10 +2576,10 @@ def assign(self, **kwargs):
         Notes
         -----
         Since ``kwargs`` is a dictionary, the order of your
-        arguments may not be preserved. To make things predicatable,
-        the columns are inserted in alphabetical order, at the end of
-        your DataFrame. Assigning multiple columns within the same
-        ``assign`` is possible, but you cannot reference other columns
+        arguments may not be preserved if you are using Python 3.5 and earlier.
+        To make things predicatable, the columns are inserted in alphabetical
+        order, at the end of your DataFrame. Assigning multiple columns within
+        the same ``assign`` is possible, but you cannot reference other columns
         created within the same ``assign`` call.
```

Comment:
I would rephrase the first couple sentences to say something like

"For python 3.6 and above, the columns are inserted in the order of **kwargs. For python 3.5 and earlier, since **kwargs is unordered, the columns are inserted in alphabetical order at the end of your DataFrame.`"

And then pick up with "Assigning multiple..."

That emphasizes that 3.6+ is good :)